components/sup/src/service_config.rs

```diff
@@ -698,6 +723,30 @@ mod test {
         assert!(re.is_match(&ip));
     }
 
+
+    #[test]
+    fn merge() {
+        let default_config = "[server]
+            port = \"8080\"
+            shutdown-port = \"8005\"
+            redirect-port = \"8443\"";
+
+        let override_config = "server = { port=\"9090\" }";
+        let default_toml = toml::Parser::new(default_config).parse().unwrap();
+        let override_toml = toml::Parser::new(override_config).parse().unwrap();
+        {
+            let result = toml_merge(&override_toml, &default_toml);
```

Comment:
Ya. I concur. This is a short term fix and there's probably goodness to be found in reworking the merge logic.